src/controller.cpp

```diff
@@ -1022,33 +1051,46 @@ bool Controller::activate(const ServerData& serverData,
               setState(StateConnecting);
 
               clearRetryCounter();
-              activateInternal(DoNotForceDNSPort, serverSelectionPolicy);
+              activateInternal(DoNotForceDNSPort, serverSelectionPolicy,
+                               initiator);
             });
 
     connect(request, &NetworkRequest::requestCompleted, this,
-            [this, serverSelectionPolicy](const QByteArray& data) {
+            [this, serverSelectionPolicy, initiator](const QByteArray& data) {
               MozillaVPN::instance()->accountChecked(data);
               setState(StateConnecting);
 
               clearRetryCounter();
-              activateInternal(DoNotForceDNSPort, serverSelectionPolicy);
+              activateInternal(DoNotForceDNSPort, serverSelectionPolicy,
+                               initiator);
             });
 
     connect(request, &QObject::destroyed, task, &QObject::deleteLater);
     return true;
   }
+  if (m_state == StateOnPartial) {
```

Comment:
Thinking out loud: Do we need to move to `StateConnecting` here since we don't care about showing any connecting UI in the client? I'm wondering if we actually need something like a `StatePartialConnecting` ? I'm thinking it could be painful to convince the front-end to not respond to `StateConnecting` here.